controllers/loadtest_controller_test.go

```diff
@@ -1,850 +1,309 @@
-			container := newRunContainer(run)
-			Expect(container.WorkingDir).To(Equal(config.WorkspaceMountPath))
-		})
-
-		It("sets image", func() {
-			image := "golang:1.14"
-			run.Image = &image
-
-			container := newRunContainer(run)
-			Expect(container.Image).To(Equal(image))
-		})
-
-		It("sets command", func() {
-			command := []string{"go"}
-			run.Command = command
-
-			container := newRunContainer(run)
-			Expect(container.Command).To(Equal(command))
-		})
-
-		It("sets args", func() {
-			command := []string{"go"}
-			args := []string{"run", "main.go"}
-			run.Command = command
-			run.Args = args
-
-			container := newRunContainer(run)
-			Expect(container.Args).To(Equal(args))
-		})
-
-		It("sets environment variables", func() {
-			env := []corev1.EnvVar{
-				{Name: "ENABLE_DEBUG", Value: "1"},
-				{Name: "VERBOSE", Value: "1"},
+			for i := range list.Items {
+				item := &list.Items[i]
+				if item.Labels[config.LoadTestLabel] == test.Name {
+					foundPodCount++
+				}
 			}
 
-			run.Env = env
-
-			container := newRunContainer(run)
-			Expect(env[0]).To(BeElementOf(container.Env))
-			Expect(env[1]).To(BeElementOf(container.Env))
-		})
+			return foundPodCount, nil
+		}).Should(Equal(expectedPodCount))
 	})
 
-	Describe("newWorkspaceVolumeMount", func() {
-		It("grants read and write access", func() {
-			volumeMount := newWorkspaceVolumeMount()
-			Expect(volumeMount.ReadOnly).To(BeFalse())
-		})
-	})
-})
-
-var _ = Describe("getRequeueTime", func() {
-	var test *grpcv1.LoadTest
-	var reconciler *LoadTestReconciler
-
-	BeforeEach(func() {
-		test = newLoadTest()
-		reconciler = &LoadTestReconciler{
-			Log: ctrl.Log.WithName("controllers").WithName("LoadTest"),
+	It("updates the status when pods terminate with errors", func() {
+		By("creating a fake environment with errored pods")
+		errorState := corev1.ContainerState{
+			Terminated: &corev1.ContainerStateTerminated{
+				ExitCode: 1,
+			},
 		}
-	})
+		builder := podbuilder.New(newDefaults(), test)
+		testSpec := &test.Spec
+		var pod *corev1.Pod
+		for i := range testSpec.Servers {
+			pod = builder.PodForServer(&testSpec.Servers[i])
+			Expect(createPod(pod, test)).To(Succeed())
+			Expect(updatePodWithContainerState(pod, errorState)).To(Succeed())
+		}
+		for i := range testSpec.Clients {
+			pod = builder.PodForClient(&testSpec.Clients[i])
+			Expect(createPod(pod, test)).To(Succeed())
+			Expect(updatePodWithContainerState(pod, errorState)).To(Succeed())
 
-	Context("when find the start time was newly filled", func() {
-		It("returns the test's timeout", func() {
-			var startTime metav1.Time
-			expectedRequeueTime := time.Duration(test.Spec.TimeoutSeconds) * time.Second
-			previousStatus := grpcv1.LoadTestStatus{}
+		}
+		if testSpec.Driver != nil {
+			pod = builder.PodForDriver(testSpec.Driver)
+			Expect(createPod(pod, test)).To(Succeed())
+			Expect(updatePodWithContainerState(pod, errorState)).To(Succeed())
+		}
```

Comment:
I think this portion of the code is saying any one of the pod that went errored will cause the entire LoadTest to be marked as errored, so we only need one errored pod, why are we creating every pods thats needed in the spec?